### PR TITLE
daemonset: added WithHostNetwork and WithVolume functions

### PR DESCRIPTION
**Pull request by ajaggapa**
Added functions to update the daemonset builder with Hostnetwork and configmap volume mount.

### CODE REVIEW

File: pkg/daemonset/daemonset.go [hunk 1]
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
 	return builder
 }
 
+// WithHostNetwork applies HostNetwork to pod's definition.
+func (builder *Builder) WithHostNetwork() *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Enabling hostnetwork to daemonset %s in namespace %s",
```
Comment (kononovn) [inline pkg/daemonset/daemonset.go:147]: Please change log message to: "Applying HostNetwork flagdaemonset %s in namespace %s"

File: pkg/daemonset/daemonset.go [hunk 2]
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
...
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.HostNetwork = true
+
+	return builder
+}
+
+// WithConfigMapVolume attaches given volume to all pod's containers.
```
Comment (kononovn) [inline pkg/daemonset/daemonset.go:155]: Please update func comment to something like:
// WithConfigMapVolume attaches given volume to all pod's containers under daemonset.

File: pkg/daemonset/daemonset.go [hunk 3]
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
...
+func (builder *Builder) WithConfigMapVolume(volumeName, mountPath string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Configuring volume %s for all pod's: %s containers. MountPath %s",
```
Comment (kononovn) [inline pkg/daemonset/daemonset.go:161]: Please adjust comment to daemonset. It's copy/paste from pod package.

File: pkg/daemonset/daemonset.go [hunk 4]
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
 	return builder
 }
 
+// WithHostNetwork applies HostNetwork to pod's definition.
```
Comment (kononovn) [inline pkg/daemonset/daemonset.go:141]: It's not pod definition. It's daemonset definition.

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix few comments above.

File: pkg/daemonset/daemonset.go (hunk 1 above)
Comment (ajaggapa) [reply pkg/daemonset/daemonset.go]: updated as suggested

File: pkg/daemonset/daemonset.go (hunk 2 above)
Comment (ajaggapa) [reply pkg/daemonset/daemonset.go]: updated as suggested

File: pkg/daemonset/daemonset.go (hunk 3 above)
Comment (ajaggapa) [reply pkg/daemonset/daemonset.go]: updated as suggested

File: pkg/daemonset/daemonset.go (hunk 4 above)
Comment (ajaggapa) [reply pkg/daemonset/daemonset.go]: updated as suggested

**Review by kononovn — CHANGES_REQUESTED** [review]
Please change PR headline and commit message to format:
package name: what was the change
Example:
sriov: added WithHostNetwork,WithConfigMapVolume  funcs

**Comment by ajaggapa** [comment]
> Please change PR headline and commit message to format: package name: what was the change Example: sriov: added WithHostNetwork,WithConfigMapVolume funcs

Updated

**Review by kononovn — DISMISSED** [review]
/lgtm

**Review by cdvultur — DISMISSED** [review]
lgtm

File: pkg/daemonset/daemonset.go [hunk 5]
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
 	return builder
 }
 
+// WithHostNetwork applies HostNetwork to daemonset definition.
+func (builder *Builder) WithHostNetwork() *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying hostnetwork flag to daemonset %s in namespace %s",
```
Comment (achuzhoy) [inline pkg/daemonset/daemonset.go:147]: Applying or Enabling?

Comment (ajaggapa) [reply pkg/daemonset/daemonset.go:147]: This gets tricky on whats makes more sense.
Started off with "Enabling", had to modify to "Applying" after @kononovn review.

File: pkg/daemonset/daemonset.go [hunk 6]
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
...
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.HostNetwork = true
+
+	return builder
+}
+
+// WithConfigMapVolume attaches given volume to all pod's containers under daemonset.
+func (builder *Builder) WithConfigMapVolume(volumeName, mountPath string) *Builder {
```
Comment (trewest) [inline pkg/daemonset/daemonset.go:156]: Can you align this method with the WithVolume deployment method?
https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment.go#L301C25-L301C35

Comment (trewest) [reply pkg/daemonset/daemonset.go:156]: For context, volumeMounts should be added through ContainerBuilder - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/pod/container.go#L353

**Review by trewest — CHANGES_REQUESTED** [review]
One comment

File: pkg/daemonset/daemonset.go (hunk 6 above)
Comment (ajaggapa) [reply pkg/daemonset/daemonset.go]: I was aligning with "WithLocalVolume" method of pod pkg. But yes, it is more structured to have separate methods for Volume and VolumeMount (container builder). 
I have made changes accordingly. PTAL.

Comment (trewest) [reply pkg/daemonset/daemonset.go]: Understood, thanks. Will make an issue for refactoring that one

File: pkg/daemonset/daemonset.go [hunk 7]
```diff
@@ -138,6 +138,51 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
...
+	glog.V(100).Infof("Enabling hostnetwork flag to daemonset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.HostNetwork = true
+
+	return builder
+}
+
+// WithConfigMapVolume defines Volume of daemonset under PodTemplateSpec with given ConfigMap.
+func (builder *Builder) WithConfigMapVolume(cmName string) *Builder {
```
Comment (trewest) [inline pkg/daemonset/daemonset.go:156]: Hey sorry for the confusion but the idea is to just make this a generic WithVolume function. We don't want to limit it to only configmaps and creating a new mutation for each volume source type would be a lot of work and difficult to manage. For the most part it is just a copy-paste of with WithVolume method from the deployment pkg

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**